fuels-abigen-macro/tests/harness.rs

```diff
@@ -1363,8 +1363,22 @@ async fn test_amount_and_asset_forwarding() {
     // Trying to forward `amount` of a different asset_id.
     // Currently, if we use this asset_id in the forwarding, the VM will
     // return `AssetIdNotFound` if amount is larger than 0.
-    let asset: [u8; 32] = id.into();
-    let call_params = CallParameters::new(None, Some(AssetId::from(asset)));
+    let asset: [u8; 32] = id.clone().into();
+
+    // withdraw some tokens to wallet
+    let c_id = testfuelcoincontract_mod::ContractId { value: id.into() };
+    let address = wallet.address();
+    let address = testfuelcoincontract_mod::Address {
+        value: address.into(),
+    };
+    instance
+        .transfer_coins_to_output(1_000_000, c_id, address)
+        .append_variable_outputs(1)
+        .call()
+        .await
+        .unwrap();
```

Comment:
we should probably add a test showing that the balance of the contract is the one we expecte

Reply:
Not that necessary to test for that. 

If there's an issue with the amount, the contract id, the address being passed, or the internal tx building process, `unwrap()` will catch it and fail the test. If the script/script_data encoding breaks, the VM will return an error. If all works, then checking if the balance of the contract is as expected would be testing the client and VM internals, which shouldn't be `fuels-rs` responsibility.

Reply:
Oh ok!!! smart